### DISCUSSION TITLE
How do I retrieve historical data from Binance using Nautilus Trader?

### DISCUSSION THREAD

**Discussion by ZzzzHeJ** [nautechsystems/nautilus_trader#1897] [Q&A] [answered]
How do I retrieve historical data from Binance using Nautilus Trader? Is there an integrated interface provided for this? If so, could you please provide a demo?

**Comment by cjdsellers** [comment, marked as answer]
Hi @ZzzzHeJ

Thanks for your interest.

So Nautilus doesn't wrap the Binance historical data API. It's assuming you would obtain data directly from Binance, or numerous other sources https://www.binance.com/en/support/faq/how-to-download-historical-market-data-on-binance-5810ae42176b4770b880ce1f14932262.

Here's an example of loading Binance order book data for Nautilus backtesting
https://nautilustrader.io/docs/latest/tutorials/backtest_binance_orderbook.

Other types such as `QuoteTick`, `TradeTick`, and `Bar` would follow a similar pattern, using the data catalog.

I hope that helps!

**Reply by ZzzzHeJ** [reply]
Thanks!
